crates/uv-trampoline/src/bounce.rs

```diff
@@ -78,7 +78,34 @@ fn make_child_cmdline() -> CString {
 
     // Only execute the trampoline again if it's a script, otherwise, just invoke Python.
     match kind {
-        TrampolineKind::Python => {}
+        TrampolineKind::Python => {
```

Comment:
Updates to the trampoline logic to set the `__PYVENV_LAUNCHER__` env var for a Python launcher (the approach taken by CPython for Python Launchers) and, if it's not a virtual environment and the `PYTHONHOME` is not set, to set the `PYTHONHOME` env var to ensure the correct directories are added to `sys.path` when running with a junction trampoline.